src/java.base/share/classes/java/net/Socket.java

```diff
@@ -1761,7 +1751,17 @@ static SocketImplFactory socketImplFactory() {
      *             {@code checkSetFactory} method doesn't allow the operation.
      * @see        java.net.SocketImplFactory#createSocketImpl()
      * @see        SecurityManager#checkSetFactory
-     */
+     * @deprecated Use {@link SocketChannel}, or subclass {@code Socket}
+     *    directly.
```

Comment:
A better suggestion in the first line would be:

```
Use a {@link SocketFactory} and subclass {@code Socket} directly.
```

Reply:
I've incorporated your suggestion now. See commit 3a89432